src/main/java/util/NoDataPathFilter.java

```diff
@@ -10,15 +10,25 @@
 import static io.restassured.RestAssured.config;
 
 public class NoDataPathFilter implements Filter {
-    private static Loader loader = new LocalLoaderImpl();
-
-
+    private static final Loader loader = new LocalLoaderImpl();
     @Override
     public Response filter(FilterableRequestSpecification filterableRequestSpecification, FilterableResponseSpecification filterableResponseSpecification, FilterContext filterContext) {
-
-        filterableRequestSpecification.given().baseUri(loader.getNoDataBasePathUrl()).config(config().sslConfig(new SSLConfig().relaxedHTTPSValidation())).header("Authorization",
+        filterableRequestSpecification.given().baseUri(returnBaseUrl()).config(config().sslConfig(new SSLConfig().relaxedHTTPSValidation())).header("Authorization",
                 "Bearer " + WriterReader.getToken());
-
         return filterContext.next(filterableRequestSpecification,filterableResponseSpecification);
     }
+
+    /**
+     * Simple method to construct the base URL with the new format following the APIG restructure
+     * @return Base url as String
+     */
+    private String returnBaseUrl() {
+        String baseUrl = loader.getNoDataBasePathUrl();
+        String branchName = loader.getBranchName();
+        if (branchName.equalsIgnoreCase("develop")) {
```

Comment:
This if statement could be shortened to use a ternary

Reply:
done